pkg/subctl/cmd/validate_cni.go

```diff
@@ -78,4 +94,123 @@ func validateCNIInCluster(clusterName string, submariner *v1alpha1.Submariner) {
 		submariner.Status.NetworkPlugin)
 	status.QueueSuccessMessage(message)
 	status.End(cli.Success)
+
+	validateCalicoIPPoolsIfCalicoCNI(config)
+}
+
+func doesCalicoConfigMapExist(clientSet kubernetes.Interface) (*v1.ConfigMap, error) {
+	cmList, err := clientSet.CoreV1().ConfigMaps(metav1.NamespaceAll).List(metav1.ListOptions{})
+	if err != nil {
+		return nil, err
+	}
+
+	for _, cm := range cmList.Items {
+		if cm.Name == "calico-config" {
+			return &cm, nil
+		}
+	}
+	return nil, nil
+}
+
+func validateCalicoIPPoolsIfCalicoCNI(config *rest.Config) {
+	dynClient, clientSet, err := getClients(config)
+	exitOnError("Error creating clients for cluster", err)
+
+	calicoConfig, _ := doesCalicoConfigMapExist(clientSet)
```

Comment:
We should  check the error here.

Reply:
There is not much we can do even if we check error. The API, doesCalicoConfigMapExist, returns calicoConfig only if its success.